internal/impl/devbox.go

```diff
@@ -836,15 +836,15 @@ func (d *Devbox) writeScriptsToFiles() error {
 	if err != nil {
 		return errors.WithStack(err)
 	}
-	written[d.scriptFilename(hooksFilename)] = struct{}{}
+	written[d.scriptPath(hooksFilename)] = struct{}{}
 
 	// Write scripts to files.
 	for name, body := range d.cfg.Shell.Scripts {
 		err = d.writeScriptFile(name, d.scriptBody(body.String()))
 		if err != nil {
 			return errors.WithStack(err)
 		}
-		written[d.scriptFilename(name)] = struct{}{}
+		written[d.scriptPath(name)] = struct{}{}
```

Comment:
@ipince these two changes have a small logic change: the map's key was filename and is not filepath. I think this is safe.